R/map.R

```diff
@@ -65,6 +65,9 @@
 #'
 #' * `walk()` returns the input `.x` (invisibly). This makes it easy to
 #'    use in a pipe. The return value of `.f()` is ignored.
+#'
+#' Any errors thrown by `.f` will be wrapped in an error with class
+#' `purrr_error_indexed`. See `vignette("programming")` for details.
```

Comment:
> Should also include a short summary of the error class invariants.
> 
> I'm thinking the `index` type should be documented as `c("numeric", "character")` even though we currently only use numeric indices?

There are a lot of cases where a non-numeric factor index is needed or a index with leading zeros.  PBkeys and spatial indexes are good examples that the index is never numeric but is ordinal like a factor.  There are still recycling issues.
However, you are correct that purr "now" uses only numeric indexes much to my disappointment.

Reply:
I think it would be nice if `index` was always an integer, and if there is also a character index available then it should go in an additional `name` slot. That is better for type stability when rethrowing errors upstream.

i.e. in tidyr I want to catch the `index` and always report it as the row number of the failure